File: Examples/Elastix/Registration/tfx.py
#!/usr/bin/env python

import SimpleITK as sitk
import sys

if len(sys.argv) < 4:
    print(f"Usage: {sys.argv[0]} <inputImage> <parameterFile> <outputImage>")
    sys.exit(1)

# Instantiate transformix
transformix_image_filter = sitk.TransformixImageFilter()

# Enable logging to console
transformix_image_filter.LogToConsoleOn()

# Read input image
moving_image = sitk.ReadImage(sys.argv[1])
transformix_image_filter.SetMovingImage(moving_image)

# Read and set transform parameter map
transform_parameter_map = sitk.ReadParameterFile(sys.argv[2])
transformix_image_filter.SetTransformParameterMap(transform_parameter_map)

# Run warp
transformix_image_filter.Execute()

# Write result image
sitk.WriteImage(transformix_image_filter.GetResultImage(), sys.argv[3])

File: Examples/Elastix/Registration/tfx.R
#=========================================================================
#
#  Copyright NumFOCUS
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#         http://www.apache.org/licenses/LICENSE-2.0.txt
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
#=========================================================================

library(SimpleITK)

args <- commandArgs(TRUE)

if (length(args) < 3) {
    stop("Usage: tfx <inputImage> <parameterFile> <outputImage>")
}

# Instantiate transformix
transformixImageFilter <- TransformixImageFilter()
transformixImageFilter$LogToConsoleOn()

# Read input
transformixImageFilter$SetMovingImage(ReadImage(args[1]))
transformixImageFilter$SetTransformParameterMap(transformixImageFilter$ReadParameterFile(args[2]))

# Run warp
resampled_image <- transformixImageFilter$Execute()

# Write result image
WriteImage(resampled_image, args[3])
